Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DayPackageSyncTool.kt

```diff
@@ -0,0 +1,142 @@
+package de.rki.coronawarnapp.diagnosiskeys.download
+
+import androidx.annotation.VisibleForTesting
+import dagger.Reusable
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.appconfig.KeyDownloadConfig
+import de.rki.coronawarnapp.diagnosiskeys.server.DiagnosisKeyServer
+import de.rki.coronawarnapp.diagnosiskeys.server.LocationCode
+import de.rki.coronawarnapp.diagnosiskeys.storage.CachedKey
+import de.rki.coronawarnapp.diagnosiskeys.storage.CachedKeyInfo.Type
+import de.rki.coronawarnapp.diagnosiskeys.storage.KeyCacheRepository
+import de.rki.coronawarnapp.storage.DeviceStorage
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDate
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.withContext
+import org.joda.time.LocalDate
+import timber.log.Timber
+import javax.inject.Inject
+
+@Reusable
+class DayPackageSyncTool @Inject constructor(
+    deviceStorage: DeviceStorage,
+    private val keyServer: DiagnosisKeyServer,
+    private val keyCache: KeyCacheRepository,
+    private val downloadTool: KeyDownloadTool,
+    private val timeStamper: TimeStamper,
+    private val configProvider: AppConfigProvider,
+    private val dispatcherProvider: DispatcherProvider
+) : BaseKeyPackageSyncTool(
+    keyCache = keyCache,
+    deviceStorage = deviceStorage,
+    tag = TAG
+) {
+
+    /**
+     * returns true if the sync was successful
+     * and false if not all files have been synced
+     */
+    internal suspend fun syncMissingDayPackages(
+        availableLocations: List<LocationCode>,
+        forceSync: Boolean
```

Comment:
Took me quite some time to fully understand what forceSync is doing and when it's set (logically).
Just from the name, I was expecting this to force the sync with CDN regardless of any caching, but that's not the case.

How about renaming this to something like `forceIndexLookup` or `forceIndexComparison` (or DayPackageIndex instead of Index for extra clarity/verbosity 😄)